scripts/archive/berkeley_migration.sh

```diff
@@ -0,0 +1,675 @@
+		echo ""
+        echo "'-t|--target-db' argument is not provided. Please provide valid connection string"
+        echo "to existing and initially migrated database. If you don't have such schema please refer to help in 'initial' subcommand" 
+		exit 1
+	fi
+
+    if [ -z $__genesis_ledger ]; then
+		echo ""
+        echo "Genesis ledger not defined"
+		exit 1
+	fi
+    if [ -z $__fork_state_hash ]; then
+		echo ""
+        echo "Fork state hash not defined. Please refer to mina or o(1) Labs team announcements regarding fork block state hash"
+        echo "which is required to run final migration"
+		exit 1
+	fi
+
+    if [ -z $__blocks_bucket ]; then
+		echo ""
+        echo "Precomputed blocks google cloud bucket not defined"
+		exit 1
+	fi
+    if [ -z $__network ]; then
+		echo ""
+        echo "Network not defined"
+		exit 1
+	fi
+    
+    run_final_migration $__batch_size \
+        $__mainnet_archive_uri \
+        $__migrated_archive_uri \
+        $__genesis_ledger \
+	    $__blocks_bucket \
+	    $__network \
+        $__fork_state_hash \
+        $__checkpoint_interval \
+        $__replayer_checkpoint
+}
+
+function run_final_migration() {
+    local __batch_size=$1
+	local __mainnet_archive_uri=$2
+	local __migrated_archive_uri=$3 
+    local __genesis_ledger=$4
+	local __blocks_bucket=$5
+	local __network=$6
+  	local __fork_state_hash=$7
+    local __checkpoint_interval=$8  
+    local __replayer_checkpoint=$9
+    
+    
+    local __date=$(date '+%Y-%m-%d_%H%M')
+    local __berkely_migration_log="berkeley_migration_$__date.log"
+    local __replayer_log="replayer_$__date.log"
+    
+    if [ ! -f $__genesis_ledger ]; then
+		echo ""
+        echo "Genesis ledger file not found: $__genesis_ledger"
+		exit 1
+	fi
+	
+    mina-berkeley-migration \
+        --mainnet-archive-uri $__mainnet_archive_uri \
+        --migrated-archive-uri $__migrated_archive_uri \
+        --batch-size $__batch_size \
+        --config-file $__genesis_ledger \
+        --blocks-bucket $__blocks_bucket \
+        --fork-state-hash $__fork_state_hash \
+        --network $__network | tee $__berkely_migration_log
+	
+    set +e # skip error because we will do validations and we are better than replayer i reporting
+
+    mina-migration-replayer \
+        --migration-mode \
+        --archive-uri $__migrated_archive_uri \
+        --input-file $__replayer_checkpoint \
+        --checkpoint-interval $__checkpoint_interval \
+        --checkpoint-file-prefix "migration" | tee $__replayer_log
+	
+    set -e # exit immediately on errors
+    
+    check_logs $__berkely_migration_log $__replayer_log
+    check_incremental_migration_progress $__replayer_checkpoint $__migrated_archive_uri 
+    check_new_replayer_checkpoints_for_incremental "migration"
+}
+
+function main(){
+    if (( ${#} == 0 )); then
+        main_help 0;
+    fi
+
+    case ${1} in
+        help | version | initial | incremental | final )
+            $1 "${@:2}";
+        ;;
```

Comment:
```suggestion
    case ${1} in
        help )
            main_help;
        version | initial | incremental | final )
            $1 "${@:2}";
        ;;
```

invokes builtin bash help otherwise